src/data/generate_synthetic_dataset_for_eval.py

```diff
@@ -0,0 +1,137 @@
+import datetime as dt
+
+import numpy as np
+import pandas as pd
+from wasabi import Printer
+
+
+def years_to_seconds(years):
+    """Calculates number of seconds in a number of years.
+
+    Args:
+        years (int): Number of years.
+
+    Returns:
+        _type_: _description_
+    """
+    return years * 365 * 24 * 60 * 60
+
+
+def return_0_with_prob(prob):
+    return 0 if np.random.random() < prob else 1
+
+
+def null_series_with_prob(
+    series: pd.Series, prob: float, null_value=np.NaN
+) -> pd.Series:
+    """Overwrite all values in series with null_value with a given probability.
+
+    Args:
+        series (pd.Series): Series.
+        prop (float): The probability of overwriting all with null_value.
+
+    Returns:
+        pd.Series:
+    """
+
+    if return_0_with_prob(prob) == 0:
+        # Replace all values in series with null_value
+        series.loc[:] = null_value
+        return series
+    else:
+        return series
+
+
+def overwrite_prop_with_null(
+    series: pd.Series, prop: float, null_value=np.NaN
+) -> pd.Series:
+    """Overwrite a proportion of all values in a series with a null value (NaN or NaT).
+
+    Args:
+        series (pd.Series): The series to overwrite in.
+        prop (float): How large a proportion to overwrite.
+    """
+    series.loc[
+        np.random.choice(series.index, int(len(series) * prop), replace=False)
+    ] = null_value
+
+    return series
+
+
+if __name__ == "__main__":
+    msg = Printer(timestamp=True)
+    base = pd.Timestamp.today()
+    n_patients = 100_000
+
+    df = pd.DataFrame()
+
+    df["dw_ek_borger"] = [np.random.randint(0, 100_000) for n in range(n_patients)]
+
+    # Generate timestamps
+    df["timestamp"] = [base for n in range(n_patients)]
```

Comment:
Hurtigere med det her?
```suggestion
    df["timestamp"] = [base] * n_patients
```